docs/components/modeler/web-modeler/git-sync.md

```diff
@@ -95,3 +95,4 @@ Existing GitHub configurations can be edited from the gear icon beside the **Syn
 - When synchronizing for the first time with a remote repository that already contains commits, Web Modeler will attempt to select a main process with a file name that matches its own main process. If there is no matching process, Web Modeler will select a process at random from the available `.bpmn` files. In the event that no `.bpmn` files exist in the remote repository, Web Modeler will not proceed, and will instead display an error message. Ensure the main process is correctly assigned, especially in cases where a random process has been selected.
 - Actions which alter the SHA of the commit to which Web Modeler is synced (for example, squash) may cause synchronization errors.
 - Timeouts may occur during a sync. In the event of a timeout, close the modal and retry the synchronization.
+- A single synchronization action is limited to incorporating a maximum of 250 commits or making changes to up to 300 files, regardless of whether these changes affect the Web Modeler files directly. Be aware that Web Modeler does not provide a notification when these thresholds are exceeded. Should you encounter this limitation, it may be necessary to initiate a fresh synchronization. This can be achieved by either changing the branch or modifying the GitHub repository URL.
```

Comment:
> Be aware that Web Modeler does not provide a notification when these thresholds are exceeded.

❓ Clarifying question for myself: Will there be an error or will this just fail silently?

> Should you encounter this limitation, it may be necessary to initiate a fresh synchronization.

Is this different from a "single synchronization action" mentioned in the beginning?

Reply:
> ❓ Clarifying question for myself: Will there be an error or will this just fail silently?

Currently, silently. I have [this issue](https://github.com/camunda/web-modeler/issues/10436) to throw an error in that case.

> Is this different from a "single synchronization action" mentioned in the beginning?

What I want to convey here is that a fresh sync is needed because then we fetch just all the files in the repository (no limitations api wise) and do not rely on incorporating changes (which would have the limitations). If that is not clear, I will make it clearer.